server/writer/writer.go

```diff
@@ -85,14 +85,17 @@ func JSON(w http.ResponseWriter, code int, v interface{}, pretty bool) {
 	}
 	headers := w.Header()
 	headers.Add("Content-Type", "application/json")
-	Bytes(w, code, bs)
+	Bytes(w, code, bs, pretty)
 }
 
 // Bytes writes a response with the specified status code and bytes.
-func Bytes(w http.ResponseWriter, code int, bs []byte) {
+func Bytes(w http.ResponseWriter, code int, bs []byte, trailingNewline bool) {
 	w.WriteHeader(code)
 	if code == 204 {
 		return
 	}
 	w.Write(bs)
+	if trailingNewline {
```

Comment:
I think this is better suited in the JSON and Error functions above. The server should be calling `JSON` in 200 cases where a body is returned and Error in 4xx/5xx cases where a body is returned. Use of `Bytes` will be for special circumstances where the response is not JSON. In those special cases, I don't think the newline param will ever get used. Note, if we make this change then we can undo a few of the modifications that were required in the server.

I think we should address this before merging.

Reply:
Changed.